packages/phoenix-evals/src/phoenix/evals/evaluators.py

```diff
@@ -174,12 +225,36 @@ def __init__(
         self,
         *,
         name: str,
-        source: SourceType,
+        source: Optional[SourceType] = None,
         direction: DirectionType = "maximize",
         input_schema: Optional[type[BaseModel]] = None,
+        kind: Optional[KindType] = None,
     ):
+        # Accept both 'kind' (preferred) and 'source' (deprecated)
+        if kind is not None and source is not None and kind != source:
+            raise ValueError("Provide only one of 'kind' or 'source' (they differ). Use 'kind'.")
+        resolved_kind: Optional[KindType] = kind if kind is not None else source
+        if source is not None and kind is None:
+            warnings.warn(
+                "Evaluator 'source' is deprecated; use 'kind' instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+        if resolved_kind is None:
+            raise ValueError("Evaluator requires 'kind'.")
+        # Heuristic deprecation and auto-migration
+        if resolved_kind == "heuristic":
+            warnings.warn(
+                "kind='heuristic' is deprecated; use kind='code' instead.",
```

Comment:
Should this warning mention that it is going to change the kind to "code" automatically as well?